### DISCUSSION TITLE
Removing polyfills from features supported for years?

### DISCUSSION THREAD

**Discussion by teoli2003** [mdn/mdn-community#568] [Content]
Hi!

We want to handle polyfills consistently on MDN.

A good first step seems to decide what features do we find useful to have (or link) to a polyfill. It doesn't mean we necessarily are actively looking for one, but we would accept a link to one. Or: _We would delete polyfills/links to polyfills for features that don't match this criterion._

I propose that:
- _we neither host, nor link to, polyfills for features that have been supported **by all browsers (in bcd) for at least 2 years**._

For Internet Explorer 11, which is still supported, although no new releases are coming out, I'm tempted to say "Goodbye, thank you for the fish!" and to anticipate by a few months (weeks?) its EoL: dealing with polyfills will not be a quick project on MDN, and this would add a lot of work to it.

(I think we shouldn't host polyfills, but this is a separate issue)

**Comment by Ryuno-Ki** [comment]
I generally agree.

For those cases, where polyfills are needed, a link to polyfill.io / https://github.com/Financial-Times/polyfill-service by @JakeChampion could be considered.

This way, the burden of maintaining the list of required polyfills could be deferred to a community that is engaged in this matter.

**Reply by JakeChampion** [reply]
I'd go with a time period of support in all major browsers as you mentioned or with a percentage of support across global users - I think a percentage of support across global users would be better as that takes into account multiple versions of a browser such as Firefox and Firefox ESR

**Comment by zloirock** [comment]
Funny, why 2 years? -) Someone still supports IE8 - I regularly receive issues about it. But the source code or polyfills definitely should be removed from MDN since too often it's incorrect or even harmful.

**Reply by teoli2003** [reply]
I do believe that supporting IE8 on the Web is not something to do, just for security reasons. MDN Web docs shouldn't make it easier. (Or we shouldn't invest money to do so. The time and work needed to do this, and therefore the costs, should be supported by the people requesting such support. 

This sounds a little bit harsh, but maintaining documentation for IE8 has a cost that is not negligible: there are hundreds of features that would need polyfills or explanation. I believe that besides IE11 (and only for about half a year), IE-only information should be a no-go on MDN.

**Reply by zloirock** [reply]
> MDN Web docs shouldn't make it easier.

Too many people are sure that it should - too often I have issues where people say that something is supported everywhere and polyfills are not required since MDN says that and they are sure that MDN data is comprehensive.

> Or we shouldn't invest money to do so.

How much money you invests for store links to proper polyfills?

One more time:

> But the source code or polyfills definitely should be removed from MDN since too often it's incorrect or even harmful.

**Comment by wbamberg** [comment]
See also https://github.com/mdn/content/issues/10072 in which @AnilSeervi is asking whether we should remove polyfills from the ES5 Array pages.

**Reply by AnilSeervi** [reply]
I'm just following up the #7556 PR where the `Array.filter()` polyfill was removed.
